### ISSUE TITLE
Hyperparameter tuning

### ISSUE THREAD

**Issue by delhomer** [Oslandia/mapillary_classification#28]
Design a experiment set to optimize the hyperparameters used during training process. This should follow the following steps:
- define which hyperparameters to tune;
- define a possible range for each of these hyperparameters;
- randomly generated a bunch of experiments (random choice is cheaper than grid choice, with a satisfying efficiency);
- run all experiments;
- choose the most accurate set of hyperparameters.

**Comment by delhomer** [comment]
Some attempts with [`hyperas` library](https://github.com/maxpumperla/hyperas). Sounds fine, excepted the fact that we do not know which parameter sets are tested. It seems impossible to test `batch_size`. I will push a `hyperparam_optimization.ipynb` notebook to show an example of such a process.

Then we can easily implement a grid search "by hand" with `itertools`:
```
for param_set in itertools.product(param1, param2, param3):
    run_model(param_set)
```
where `param1`, `param2` and `param3` are lists of possible values for each focused parameters. For this try I can push a new python module to illustrate such a treatment.

**Comment by delhomer** [comment]
Will be fixed by PR #66 .

**Comment by delhomer** [comment]
PR #66 is now merged, the code is ready for tuning hyperparameters. I close the issue.
